File: src/java/com/kitware/pulse/cdm/patient/actions/SECardiovascularMechanicsModification.java
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

package com.kitware.pulse.cdm.patient.actions;

import com.kitware.pulse.cdm.bind.PatientActions.CardiovascularMechanicsModificationData;
import com.kitware.pulse.cdm.system.physiology.SECardiovascularMechanicsModifiers;

public class SECardiovascularMechanicsModification extends SEPatientAction
  {
    protected SECardiovascularMechanicsModifiers modifiers = null;
    protected String modifiersFile = "";
    protected boolean restabilize = true;

    public SECardiovascularMechanicsModification()
    {
      super();
    }

    public SECardiovascularMechanicsModification(SECardiovascularMechanicsModification other)
    {
      copy(other);
    }

    public void copy(SECardiovascularMechanicsModification other)
    {
      super.copy(other);
      if (other.modifiers != null)
        this.getModifiers().copy(other.modifiers);
      this.modifiersFile = other.modifiersFile;
      this.restabilize = other.restabilize;
    }

    public void clear()
    {
      super.clear();
      if (this.modifiers != null)
        this.modifiers.clear();
      this.modifiersFile = "";
      this.restabilize = true;
    }
    
    public static void load(CardiovascularMechanicsModificationData src, SECardiovascularMechanicsModification dst)
    {
      if (src.hasModifiers())
        SECardiovascularMechanicsModifiers.load(src.getModifiers(), dst.getModifiers());
      else if (src.getModifiersFile().isEmpty())
        dst.setModifiersFile(src.getModifiersFile());
      dst.restabilize = src.getRestabilize();
    }
    public static CardiovascularMechanicsModificationData unload(SECardiovascularMechanicsModification src)
    {
      CardiovascularMechanicsModificationData.Builder dst = CardiovascularMechanicsModificationData.newBuilder();
      unload(src,dst);
      return dst.build();
    }
    protected static void unload(SECardiovascularMechanicsModification src, CardiovascularMechanicsModificationData.Builder dst)
    {
      if(src.hasModifiers())
        dst.setModifiers(SECardiovascularMechanicsModifiers.unload(src.getModifiers()));
      else if (src.hasModifiersFile())
        dst.setModifiersFile(src.getModifiersFile());
      dst.setRestabilize(src.restabilize);
    }

    public boolean isValid()
    {
      return hasModifiers() || hasModifiersFile();
    }
    
    public boolean restabilize() { return this.restabilize; }
    void setRestabilization(boolean b) { this.restabilize = b; }

    public boolean hasModifiers()
    {
      return this.modifiers != null;
    }
    public SECardiovascularMechanicsModifiers getModifiers()
    {
      if (this.modifiers == null)
        this.modifiers = new SECardiovascularMechanicsModifiers();
      return this.modifiers;
    }

    public boolean hasModifiersFile()
    {
      return this.modifiersFile != null && !this.modifiersFile.isEmpty();
    }
    public String getModifiersFile()
    {
      return this.modifiersFile;
    }
    public void setModifiersFile(String s)
    {
      this.modifiersFile = s;
    }

    public String toString()
    {
      String str = "Cardiovascular Mechanics Modification";
      if (this.hasModifiersFile())
        str += "\n\tModifiers File: " + this.modifiersFile;
      else if (hasModifiers())
      {
        str += modifiers.toString();
      }
      str += "\n\tRestablize: " + this.restabilize;
      return str;
    }
  }

File: src/csharp/pulse/cdm/patient/actions/SECardiovascularMechanicsModification.cs
﻿/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

namespace Pulse.CDM
{
  public class SECardiovascularMechanicsModification : SEPatientAction
  {
    protected SECardiovascularMechanicsModifiers modifiers = null;
    protected string modifiersFile = "";
    protected bool restabilize = true;

    public SECardiovascularMechanicsModification()
    {

    }

    public SECardiovascularMechanicsModification(SECardiovascularMechanicsModification other)
    {
      Copy(other);
    }

    public void Copy(SECardiovascularMechanicsModification other)
    {
      base.Copy(other);
      if (other.modifiers != null)
        this.GetModifiers().Copy(other.modifiers);
      this.modifiersFile = other.modifiersFile;
      this.restabilize = other.restabilize;
    }

    public override void Clear()
    {
      base.Clear();
      if (this.modifiers != null)
        this.modifiers.Clear();
      this.modifiersFile = "";
      restabilize = true;
    }

    public override bool IsValid()
    {
      return HasModifiers() || HasModifiersFile();
    }

    public bool Restabilize() { return restabilize; }
    public void SetRestabilization(bool b) { restabilize = b; }

    public bool HasModifiers()
    {
      return this.modifiers != null;
    }
    public SECardiovascularMechanicsModifiers GetModifiers()
    {
      if (this.modifiers == null)
        this.modifiers = new SECardiovascularMechanicsModifiers();
      return this.modifiers;
    }

    public bool HasModifiersFile()
    {
      return !string.IsNullOrEmpty(this.modifiersFile);
    }
    public string GetModifiersFile()
    {
      return this.modifiersFile;
    }
    public void SetModificationFile(string s)
    {
      this.modifiersFile = s;
    }

    public override string ToString()
    {
      string str = "Cardiovascular Mechanics Modification";
      if (this.HasModifiersFile())
        str += "\n\tModifiers File: " + this.modifiersFile;
      else if (HasModifiers())
      {
        str += modifiers.ToString();
      }
      str += "\n\tRestabilize: " + this.restabilize;
      return str;
    }
  }
}
